lib/ecto/query/builder.ex

```diff
@@ -74,8 +74,8 @@ defmodule Ecto.Query.Builder do
   with `^index` in the query where index is a number indexing into the
   map.
   """
-  @spec escape(Macro.t, quoted_type | {:in, quoted_type} | {:out, quoted_type}, {list, acc},
-               Keyword.t, Macro.Env.t | {Macro.Env.t, fun}) :: {Macro.t, {list, acc}}
+  @spec escape(Macro.t, quoted_type | {:in, quoted_type} | {:out, quoted_type} | {:splice, quoted_type},
+               {list, acc}, Keyword.t, Macro.Env.t | {Macro.Env.t, fun}) :: {Macro.t, {list, acc}}
```

Comment:
Question, could we reuse `:in` instead of introducing `:splice`? It seems they behave the same inside the planner when it comes to parameters. In the AST they can still be separate nodes.

Reply:
Oh, maybe a reason why we can't unify is because PG and MySQL will treat them differently, so ignore me.

Reply:
> Oh, maybe a reason why we can't unify is because PG and MySQL will treat them differently
Yeah exactly this. At first I wanted to do the same thing but then postgres was removing the `:in`.